src/lib/application-state-stores/explorer-stores.ts

```diff
@@ -0,0 +1,81 @@
+import { writable } from "svelte/store";
+import type { MetricsExplorerEntity } from "$lib/redux-store/explore/explore-slice";
+import type { MetricViewMetaResponse } from "$common/rill-developer-service/MetricViewActions";
+import type { TimeSeriesTimeRange } from "$common/database-service/DatabaseTimeSeriesActions";
+
+export interface MetricsExplorerStoreType {
+  entities: Record<string, MetricsExplorerEntity>;
+}
+export const MetricsExplorerStore = writable({
+  entities: {},
+} as MetricsExplorerStoreType);
+
+const UpdateMetricsExplorer = (
+  id: string,
+  callback: (metricsExplorer: MetricsExplorerEntity) => void,
+  absenceCallback?: () => MetricsExplorerEntity
+) => {
+  MetricsExplorerStore.update((state) => {
+    if (!state.entities[id]) {
+      if (absenceCallback) {
+        state.entities[id] = absenceCallback();
+      }
+      return state;
+    }
+    callback(state.entities[id]);
+    return state;
+  });
+};
+
+export const syncMetricsExplorer = (
```

Comment:
This `sync` code is temporary, right? We'll remove it later?

Reply:
We will need to make sure we remove measures and dimensions that is in the store. The sync will take care of it.